### PR TITLE
Separate Event Manager to doctrine/event-manager

### PR DESCRIPTION
**Pull request by Majkl578**
EventManager + EventSubscriber + EventArgs separated into [doctrine/event-manager](https://github.com/doctrine/event-manager).

One step closer to finishing #826.

_Note: `@dev` needs to be removed before 2.9.0 (once `1.0.0` is tagged)._

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -19,7 +19,8 @@
         "doctrine/cache": "1.*",
         "doctrine/collections": "1.*",
         "doctrine/lexer": "1.*",
-        "doctrine/annotations": "1.*"
+        "doctrine/annotations": "1.*",
+        "doctrine/event-manager": "1.*@dev"
```
Comment (alcaeus) [inline composer.json]: `^1.0.0@dev` (without `@dev` once we release it)?

Comment (Majkl578) [reply composer.json]: I replaced all constraints with caret. 👍

Comment (Majkl578) [reply composer.json]: bumped :+1:

**Comment by Ocramius** [comment]
:steam_locomotive:

**Pull request merged**